src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -72,9 +85,11 @@ internal abstract class AbstractRecommendationServiceBasedCompletionProvider<TSy
 
         private static bool IsValidForTaskLikeTypeOnlyContext(ISymbol symbol, TSyntaxContext context)
         {
-            // We want to allow all namespaces as the user may be typing a namespace name to get to a task-like type.
-            if (symbol.IsNamespace())
-                return true;
+            if (symbol is IAliasSymbol alias)
+                symbol = alias.Target;
+
+            if (symbol is INamespaceSymbol @namespace)
+                return @namespace.GetMembers().Any(m => IsValidForTaskLikeTypeOnlyContext(m, context));
```

Comment:
This check is likely not ok. It can recurse and walk the entire namespace tree. That's a lot of symbol loading.